google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java

```diff
@@ -310,88 +313,36 @@ public void testGetBlobFailNonExistingGeneration() {
-            .setMetadata(metadata)
-            .build();
-    BlobInfo blob2 =
-        BlobInfo.newBuilder(bucket, blobNames[1])
-            .setContentType(CONTENT_TYPE)
-            .setMetadata(metadata)
-            .build();
+    BlobInfo blob1 = BlobInfo.newBuilder(bucket, name1).setMetadata(metadata).build();
+    BlobInfo blob2 = BlobInfo.newBuilder(bucket, name2).setMetadata(metadata).build();
     Blob remoteBlob1 = storage.create(blob1);
     Blob remoteBlob2 = storage.create(blob2);
-    assertNotNull(remoteBlob1);
-    assertNotNull(remoteBlob2);
+
+    ImmutableSet<Map<String, String>> expected =
+        Stream.of(remoteBlob1, remoteBlob2)
+            .map(BlobInfo::getMetadata)
+            .collect(ImmutableSet.toImmutableSet());
+
     Page<Blob> page =
         storage.list(
             bucket.getName(),
-            Storage.BlobListOption.prefix("test-list-blobs-selected-fields-blob"),
+            Storage.BlobListOption.prefix(baseName),
             Storage.BlobListOption.fields(BlobField.METADATA));
-    // Listing blobs is eventually consistent, we loop until the list is of the expected size. The
-    // test fails if timeout is reached.
-    while (Iterators.size(page.iterateAll().iterator()) != 2) {
-      Thread.sleep(500);
-      page =
-          storage.list(
-              bucket.getName(),
-              Storage.BlobListOption.prefix("test-list-blobs-selected-fields-blob"),
-              Storage.BlobListOption.fields(BlobField.METADATA));
-    }
-    Set<String> blobSet = ImmutableSet.of(blobNames[0], blobNames[1]);
-    Iterator<Blob> iterator = page.iterateAll().iterator();
-    while (iterator.hasNext()) {
-      Blob remoteBlob = iterator.next();
-      assertEquals(bucket.getName(), remoteBlob.getBucket());
-      assertTrue(blobSet.contains(remoteBlob.getName()));
-      assertEquals(metadata, remoteBlob.getMetadata());
-      assertNull(remoteBlob.getContentType());
-    }
-  }
 
-  @Test(timeout = 5000)
-  public void testListBlobsEmptySelectedFields() throws InterruptedException {
+    ImmutableSet<Blob> blobs = ImmutableSet.copyOf(page.iterateAll());
 
-    String[] blobNames = {
-      "test-list-blobs-empty-selected-fields-blob1", "test-list-blobs-empty-selected-fields-blob2"
-    };
-    BlobInfo blob1 = BlobInfo.newBuilder(bucket, blobNames[0]).setContentType(CONTENT_TYPE).build();
-    BlobInfo blob2 = BlobInfo.newBuilder(bucket, blobNames[1]).setContentType(CONTENT_TYPE).build();
-    Blob remoteBlob1 = storage.create(blob1);
-    Blob remoteBlob2 = storage.create(blob2);
-    assertNotNull(remoteBlob1);
-    assertNotNull(remoteBlob2);
-    Page<Blob> page =
-        storage.list(
-            bucket.getName(),
-            Storage.BlobListOption.prefix("test-list-blobs-empty-selected-fields-blob"),
-            Storage.BlobListOption.fields());
-    // Listing blobs is eventually consistent, we loop until the list is of the expected size. The
-    // test fails if timeout is reached.
```

Comment:
Listing objects is strongly consistent[1], we don't need the manual retry loop here.

[1] https://cloud.google.com/storage/docs/consistency#strongly_consistent_operations